lucene/highlighter/src/java/org/apache/lucene/search/uhighlight/MemoryIndexOffsetStrategy.java

```diff
@@ -72,9 +73,6 @@ private static CharacterRunAutomaton buildCombinedAutomaton(UHComponents compone
           MultiTermHighlighting.extractAutomata(spanQuery, components.getFieldMatcher(), true));//true==lookInSpan
     }
 
-    if (allAutomata.size() == 1) {
```

Comment:
Can't you make this method, buildCombinedAutomaton, return a CharArrayMatcher now?  Then you could retain this optimization for a single automata, and also simplify the last anonymous implementation to *not* be a CharacterRunAutomaton.

Reply:
done